internal/utils/utils.go

```diff
@@ -17,20 +17,29 @@ package utils
 import (
 	"context"
 	"fmt"
+	"regexp"
+	"strings"
 	"time"
 
 	"github.com/cilium/cilium-cli/defaults"
 )
 
+func CheckVersion(version string) bool {
+	var check = regexp.MustCompile(`^([v]|0|[1-9][0-9]*\.)?(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)(-[a-zA-Z0-9]+)*\.*(0|[1-9][0-9]*)?`).MatchString
+	return check(version)
+}
+
 func BuildImagePath(userImage, defaultImage, userVersion, defaultVersion string) string {
 	if userImage == "" {
 		userImage = defaultImage
 	}
 
-	if userVersion == "" {
+	switch {
+	case userVersion == "":
 		userVersion = defaultVersion
+	case !strings.HasPrefix(userVersion, "v"):
+		userVersion = "v" + userVersion
```

Comment:
If a developer passes `--agent-image docker.io/alice/cilium`  and `--version my-custom-version`, will this produce `docker.io/alice/cilium:vmy-custom-version`?

I'm wondering if we should actually do this `v` prepend earlier as well, at the same place where we choose whether to perform the regex validation or not.

Reply:
Good catch fixed. I realized another thing as there is only one `--version` command in https://github.com/cilium/cilium-cli/blob/373236d85ef4812f9602b7f449dbfa56dc527675/install/install.go#L1034 however in the same place  there are two different commands for `--agent-image` and ` --operator-image`.   Can `cilium agent` and `cilium operator` run in different versions , if so shouldn't we support this case ?

Reply:
I wouldn't worry too much about handling that case for now. For most users, it will be one consistent version like `v1.10.0`. For developers, they will often specify an image built from the CI which should use a consistent tag as well I think.